### PR TITLE
[WIP-Coverage-failed-oneDNN] enable hogwild for oneDNN

### PR DESCRIPTION
**Pull request by jczaja**
### PR types
Others

### PR changes
Others

### Describe
Hogwild workers were not having possiblity to be executed using oneDNN kernels. So this PR is adding such an option.

### CODE REVIEW

**Review by arlesniak — DISMISSED** [review]
LGTM

**Review by arlesniak — DISMISSED** [review]
LGTM

**Comment by jczaja** [comment]
@chenwhql   Please review and id positive approve CI-APPROVAL

File: paddle/fluid/framework/hogwild_worker.cc [hunk 1]
```diff
@@ -112,6 +117,20 @@ void HogwildWorker::BindingDataFeedMemory() {
...
+    for (size_t bid = 0; bid < main_prog.Size(); ++bid) {
+      auto *block = const_cast<ProgramDesc &>(main_prog).MutableBlock(bid);
+      for (auto *op : block->AllOps()) {
+        if (op->HasAttr("use_mkldnn")) {
+          op->SetAttr("use_mkldnn", true);
+        }
+      }
+    }
+  };
+  if (FLAGS_use_mkldnn) EnableMKLDNN();
```
Comment (arogowie-intel) [inline paddle/fluid/framework/hogwild_worker.cc]: Why using lambda and not just put it's body into the `if` body?
```suggestion
  if (FLAGS_use_mkldnn) {
    VLOG(3) << "use_mkldnn=True";
    for (size_t bid = 0; bid < main_prog.Size(); ++bid) {
      auto *block = const_cast<ProgramDesc &>(main_prog).MutableBlock(bid);
      for (auto *op : block->AllOps()) {
        if (op->HasAttr("use_mkldnn")) {
          op->SetAttr("use_mkldnn", true);
        }
      }
    }
   }
  };
```

File: paddle/fluid/framework/executor.cc [hunk 2]
```diff
@@ -168,8 +170,8 @@ void Executor::Run(const ProgramDesc& pdesc, Scope* scope, int block_id,
                    const std::vector<std::string>& skip_ref_cnt_vars,
                    bool force_disable_gc, bool keep_kid_scopes) {
   platform::RecordBlock b(block_id);
-  if (FLAGS_use_mkldnn) EnableMKLDNN(pdesc);
 #ifdef PADDLE_WITH_MKLDNN
+  if (FLAGS_use_mkldnn) EnableMKLDNN(pdesc);
```
Comment (arogowie-intel) [inline paddle/fluid/framework/executor.cc]: If you're putting `EnableMKLDNN` under `#ifdef` then you can remove the `ifdef`s from this function https://github.com/PaddlePaddle/Paddle/blob/ec2b0e14909cec83797da434e9cfbff27d11dc20/paddle/fluid/framework/executor.cc#L577

File: paddle/fluid/framework/executor.cc [hunk 3]
```diff
@@ -296,8 +298,8 @@ void Executor::Run(const ProgramDesc& program, Scope* scope,
                    const std::string& feed_holder_name,
                    const std::string& fetch_holder_name) {
   platform::RecordBlock b(kProgramId);
-  if (FLAGS_use_mkldnn) EnableMKLDNN(program);
 #ifdef PADDLE_WITH_MKLDNN
+  if (FLAGS_use_mkldnn) EnableMKLDNN(program);
```
Comment (wzzju) [inline paddle/fluid/framework/executor.cc]: The original codes may be correct.

File: paddle/fluid/framework/hogwild_worker.cc (hunk 1 above)
Comment (jczaja) [reply paddle/fluid/framework/hogwild_worker.cc]: ok

File: paddle/fluid/framework/executor.cc (hunk 3 above)
Comment (jczaja) [reply paddle/fluid/framework/executor.cc]: ok

**Pull request closed without merging**